cli/src/commands/git/colocate.rs

```diff
@@ -0,0 +1,304 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Write as _;
+use std::path::Path;
+
+use jj_lib::file_util::IoResultExt as _;
+use jj_lib::git;
+use jj_lib::repo::Repo as _;
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::WorkspaceCommandHelper;
+use crate::command_error::CommandError;
+use crate::command_error::user_error;
+use crate::command_error::user_error_with_message;
+use crate::git_util::is_colocated_git_workspace;
+use crate::ui::Ui;
+
+/// Manage Jujutsu repository colocation with Git
+#[derive(clap::Args, Clone, Debug)]
+pub struct GitColocateArgs {
+    /// Enable colocation (convert into a colocated Jujutsu/Git repository)
+    ///
+    /// This moves the underlying Git repository that is found inside the .jj
+    /// directory to the root of the Jujutsu workspace. This allows you to
+    /// use Git commands directly in the Jujutsu workspace.
+    #[arg(long, conflicts_with = "disable")]
+    enable: bool,
+
+    /// Disable colocation (convert into a non-colocated Jujutsu/Git
+    /// repository)
+    ///
+    /// This moves the Git repository that is at the root of the Jujutsu
+    /// workspace into the .jj directory. Once this is done you will no longer
+    /// be able to use Git commands directly in the Jujutsu workspace.
+    #[arg(long, conflicts_with = "enable")]
+    disable: bool,
+}
+
+pub fn cmd_git_colocate(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &GitColocateArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+
+    // Check if backend is Git
+    if git::get_git_backend(workspace_command.repo().store()).is_err() {
+        return Err(user_error(
+            "This command requires a repository backed by Git. This repository appears to be \
+             using a different backend.",
+        ));
+    }
+
+    // Ensure that this is the main workspace
+    let workspace_root = workspace_command.workspace_root();
+    let jj_repo_path = workspace_root.join(".jj").join("repo");
+    if jj_repo_path.is_file() {
+        return Err(user_error(
+            "This command cannot be used in a non-main Jujutsu workspace.",
+        ));
+    }
+
+    if args.enable {
+        enable_repository_colocation(ui, &mut workspace_command)
+    } else if args.disable {
+        disable_repository_colocation(ui, &mut workspace_command)
+    } else {
+        show_status(ui, &mut workspace_command)
+    }
```

Comment:
nit: Add different command for status output? We generally avoid overloading a single command for mutation and status output. I think it's important here because `jj git colocate --enable|--disable` is a destructive operation, which the user shouldn't run casually.

Reply:
As suggested by you and others, I've changed the `--enable` and `--disable` flags into separate sub-commands. I've also added a `status` sub-command that must be used to get the current colocation status. Calling `jj git colocate` will just show the help message.